src/Routing/Middleware/RoutingMiddleware.php

```diff
@@ -55,7 +56,14 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
                 $response->getHeaders()
             );
         }
+        $middleware = Router::getMatchingMiddleware($request->getUri()->getPath());
```

Comment:
Router doesn't have a way to get the collection right now. I could change that though. It might be nice if the middleware could have a route collection passed as a constructor parameter.